lib/HeroBonus.cpp

```diff
@@ -1198,27 +1222,36 @@ std::string Bonus::Description() const
 	std::ostringstream str;
 
 	if(description.empty())
-		switch(source)
+	{
+		if(stacking.empty() || stacking == "ALWAYS")

```

Comment:
Does this change hint that `stacking` is actually new special kind of bonus source?

Reply:
Yes, the stacking string now shows up as the source:
![Morale Info](https://i.imgur.com/eHC99eY.png)

Reply:
I mean can we reuse bonus source/ source id field for that and do not add stacking field?

Reply:
No, bonus source is used for other things, e.g. we might have different secondary skills granting non-stacking bonuses, but setting them to same source id means we can't remove them easily when upgrading to the next level. Also bonus source is set automatically throughout the code, and having it identify the actual source is helpful for debugging. And finally we may want bonuses from different source types to not stack.